enterprise_gateway/services/kernels/handlers.py

```diff
@@ -45,35 +47,40 @@ async def post(self):
             if len(kernels) >= max_kernels:
                 raise tornado.web.HTTPError(403, "Resource Limit")
 
-        # Try to get env vars from the request body
+        env = {}
+        # Try to get tenant_id and env vars from the request body
         model = self.get_json_body()
-        if model is not None and "env" in model:
-            if not isinstance(model["env"], dict):
-                raise tornado.web.HTTPError(400)
-            # Start with the PATH from the current env. Do not provide the entire environment
-            # which might contain server secrets that should not be passed to kernels.
-            env = {"PATH": os.getenv("PATH", "")}
-            # Whitelist environment variables from current process environment
-            env.update(
-                {
-                    key: value
-                    for key, value in os.environ.items()
-                    if key in self.env_process_whitelist
-                }
-            )
-            # Whitelist KERNEL_* args and those allowed by configuration from client.  If all
-            # envs are requested, just use the keys from the payload.
-            env_whitelist = self.env_whitelist
-            if env_whitelist == ["*"]:
-                env_whitelist = model["env"].keys()
-            env.update(
-                {
-                    key: value
-                    for key, value in model["env"].items()
-                    if key.startswith("KERNEL_") or key in env_whitelist
-                }
-            )
+        if model is not None:
+            tenant_id = model.get("tenant_id", UNIVERSAL_TENANT_ID)
```

Comment:
is `tenant_id` going to be an `env` variable which will be set on jupyterlab / notebook server?  If yes, may be this should be passed in the "env" part of the payload?

Reply:
It will be a configurable in `GatewayClient` which _can_ be set using `JUPYTER_GATEWAY_TENANT_ID` but that's a separate thing.  We can certainly require `JUPYTER_GATEWAY_TENANT_ID` reside in the `env` stanza - independent of how it's configured.  This will also avoid an API change (per se) in the start request.

So, if we say tenant_id is conveyed via the env portion of the body, we _could_ use `KERNEL_TENANT_ID` as the lookup key.  We'd still need to handle the case where no `env` is used (b/c) and set `KERNEL_TENANT_ID` to `UNIVERSAL_TENANT_ID` however.

I'm good making this change assuming you're okay with the above.

Reply:
Per discussion, the tenant_id will be conveyed in the `env` body via `JUPYTER_GATEWAY_TENANT_ID` and _converted_ to `KERNEL_TENANT_ID` _within_ EG.  Requests not including the tenant_id in the `env` will have `KERNEL_TENANT_ID` set to `UNIVERSAL_TENANT_ID`.